pymongo/monitoring.py

```diff
@@ -652,7 +666,9 @@ def command(self) -> _DocumentOut:
     @property
     def database_name(self) -> str:
         """The name of the database this command was run against."""
-        return self.__db
+        db = super().database_name
+        assert db is not None
```

Comment:
Why do we need the assert?

Reply:
Our typechecker had been raising issues because `self.__db` is actually typed as `Optional[str]`. Used an assert to mitigate that error. 

Alternatively, I could just `# type: ignore`